File: 11. Container With Most Water.cpp
class Solution {
public:
    int max(int a, int b)
    {
        return a > b ? a : b;
    }
    int min(int a, int b)
    {
        return a < b ? a : b;
    }
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int res = 0;
        while (l < r)
        {
            res = max(res, min(height[l], height[r]) * (r-l));
            if (height[l] < height[r])
                l++;
            else if (height[l] > height[r])
                r--;
            else
            {
                l++; r--;
            }
        }
        return res;
    }
};

File: 11. Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        l = 0
        r = len(height)-1
        res = min(height[l], height[r]) * (r-l)
        while l < r:
            res = max(res, min(height[l], height[r]) * (r-l))
            if height[l] < height[r]:
                l += 1
            elif height[r] < height[l]:
                r -= 1
            else:
                l += 1
                r -= 1
        return res
